### PR TITLE
chore: fix ts build

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did
futz with the typescript configs and fix some typescript errors in various test files

should unblock release (see https://github.com/patternfly/patternfly-elements/actions/runs/3780052452/jobs/6425817091)

this was cherry-picked from #2283

### CODE REVIEW

File: tools/pfe-tools/tsconfig.json [hunk 1]
```diff
@@ -7,14 +7,8 @@
     "dev-server/bundle.ts"
   ],
   "compilerOptions": {
-    "target": "es2022",
```
Comment (heyMP) [inline tools/pfe-tools/tsconfig.json:10]: We had to bump this to es2022 to support `Array.at`.  Do you think this will cause a regression at all?

Comment (bennypowers) [reply tools/pfe-tools/tsconfig.json:10]: Might. Let's put it back

File: tools/pfe-tools/tsconfig.json [hunk 2]
```diff
@@ -7,14 +7,8 @@
     "dev-server/bundle.ts"
   ],
   "compilerOptions": {
```
Comment (bennypowers) [inline tools/pfe-tools/tsconfig.json:9]:
```suggestion
  "compilerOptions": {
    "target": "es2022",
```

Comment (bennypowers) [inline tools/pfe-tools/tsconfig.json:9]:
```suggestion
  "compilerOptions": {
    "target": "es2022",
```

**Pull request merged**